### ISSUE TITLE
Server actions borked when setting generateStaticParams in layout

### ISSUE THREAD

**Issue by ravinggenius** [vercel/next.js#54992]
### Link to the code that reproduces this issue or a replay of the bug

https://github.com/ravinggenius/i18next-next13-server-actions

### To Reproduce

0. `npm install`
1. `npm run dev`
2. Open http://localhost:3000/
3. Fill in arbitrary email and password
4. Submit form

### Current vs. Expected behavior

I expect to see a console.log output of the credentials in the terminal window that is running the development server. The browser should also be redirected to a page containing a brief welcome message.

Instead the server action (`createSession` defined in `src/app/[locale]/actions.ts`) is never called. The browser receives a `405 method not allowed` response.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP PREEMPT Tue Aug 8 19:05:00 PDT 2023
    Binaries:
      Node: 18.7.0
      npm: 8.15.0
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.20-canary.16
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
    Next.js Config:
      output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Internationalization (i18n)

### Additional context

I initially thought this was [an issue](https://github.com/i18next/next-13-app-dir-i18next-example/issues/29) with a tutorial I followed to get `i18next` working. They suggested the issue is related to https://github.com/vercel/next.js/issues/49408 instead. I went through that thread and tried the popular suggestion (add `export const dynamic = "force-dynamic";`), but it didn't fix my issue.

I also tried commenting out `export const generateStaticParams ...`. This let me trigger the server action and see the welcome message, but it somehow blocks the session cookie from being set.

**Comment by irg1008** [comment]
It happens to me as well, did you find any work around or we are stuck and cannot have both server actions and i18n?

**Comment by irg1008** [comment]
Here someone says the fixed it by splitting layouts: https://github.com/vercel/next.js/issues/49408#issuecomment-1703337599
But I don't see how that would work? 🤔

**Comment by sangdth** [comment]
I faced this issue too, and I tried to use the force-dynamic, it does not help too. Tried to follow the comment about splitting but as long as there is `generateStaticParams` inside the `/[locale]/layout`, it causes the server action 405 issue.

**Comment by ravinggenius** [comment]
@irg1008 my work-around is to kinda punt. I'm rolling my own i18n solution with a similar API to `react-i18next`. I didn't move my routes to `app/[locale]/*` though. Instead I'm using cookie/`intl-parse-accept-language` to look up the locale on the server and cookie/`preferred-locale` to look up the locale client-side. Everything works fine on the server, but I'm having some trouble hooking into `Suspense` on the client. I think it should work as a stop-gap though.

**Comment by ravinggenius** [comment]
If you're interested, here's where I ended up: https://github.com/ravinggenius/i18next-next13-server-actions/tree/localization-basic. Check out the files in `src/i18n`. The server-side seems to work fine. The client-side works, but there's always a flash of the translation keys prior to the translation showing. I tried solving that in another branch (`localization-basic-USE`), but it just wasn't working. (If somebody knows how I can get `use()` and `Suspense` to work, I'll be very grateful.)

For my purposes this solution is good enough. I hope to upgrade to `i18next` at some point, as there's a bunch of features I don't want to re-implement.

**Comment by irg1008** [comment]
I think they fixed the issue: https://github.com/vercel/next.js/issues/49408#issuecomment-1720291375

They are merging this on Next.js 13.5 next week

**Comment by irg1008** [comment]
The just dropped Next.js 13.5. The 405 issues doesn't happen anymore but another equally annoying happens when using i18n. Maybe it happens to you as well:

When submitting a form, then calling server action, it won't call the server action while behind a locale that's not default.

In my case I have `en` as default and it works as it should. Then with any other default the first submit gets ignored, and then the server action is called twice

**Comment by ravinggenius** [comment]
I wasn't able to see it working as expected even after updating to Next 13.5.2. However I also gave up as I never really wanted to maintain the locale in the path, and I found a cleaner solution.

Here's my [preferred solution](https://github.com/ravinggenius/minecraft-tools/tree/main/src/i18n), with the locale set by examining the `Accept-Language` header (server) or interrogating the browser (client). Both use a (shared) cookie to override the locale if needed.
